core/src/main/java/nl/nn/adapterframework/doc/model/FrankElement.java

```diff
@@ -0,0 +1,249 @@
+				&& method.getReturnType().getName().equals("void")
+				&& (method.getParameterTypes().length == 1)
+				&& (method.getParameterTypes()[0].isPrimitive()
+						|| method.getParameterTypes()[0].getName().equals(JAVA_STRING));
+		boolean isGetter = (
+					method.getReturnType().isPrimitive()
+					|| method.getReturnType().getName().equals(JAVA_STRING)
+				) && (method.getParameterTypes().length == 0);
+		return isSetter || isGetter;
+	}
+
+	private void compareGetterWithSetter(Method setter, Method getter) {
+		String setterType = setter.getParameterTypes()[0].getName();
+		String getterType = getter.getReturnType().getName();
+		if(! getterType.equals(setterType)) {
+			log.warn(String.format("In Frank element %s: setter %s has type %s while the getter has type %s",
+					simpleName, setter.getName(), setterType, getterType));
+		}
+	}
+
+	private void documentAttribute(
+			FrankAttribute attribute,
+			Method method,
+			FrankElementStore store) {
+		attribute.setDeprecated(AnnotationUtils.findAnnotation(method, Deprecated.class) != null);
+		IbisDocRef ibisDocRef = AnnotationUtils.findAnnotation(method, IbisDocRef.class);
+		if(ibisDocRef != null) {
+			ParsedIbisDocRef parsed = parseIbisDocRef(ibisDocRef, method);
+			IbisDoc ibisDoc = AnnotationUtils.findAnnotation(parsed.getReferredMethod(), IbisDoc.class);
+			if(ibisDoc != null) {
+				attribute.setDescribingElement(frankElement(
+						parsed.getReferredMethod().getDeclaringClass(), store));
+				ibisDocIntoFrankElement(ibisDoc, attribute);
+				if(parsed.hasOrder) {
+					attribute.setOrder(parsed.getOrder());
+				}
+				return;
+			}
+		}
+		IbisDoc ibisDoc = AnnotationUtils.findAnnotation(method, IbisDoc.class);
+		if(ibisDoc != null) {
+			ibisDocIntoFrankElement(ibisDoc, attribute);
+		}
+		else {
+			log.warn(String.format("No documentation available for FrankElement %s, attribute %s",
+					simpleName, attribute.getName()));
+		}
+	}
+
+	private class ParsedIbisDocRef {
+		private @Getter @Setter boolean hasOrder;
+		private @Getter @Setter int order;
+		private @Getter @Setter Method referredMethod;
+	}
+
+	private ParsedIbisDocRef parseIbisDocRef(IbisDocRef ibisDocRef, Method originalMethod) {
+		ParsedIbisDocRef result = new ParsedIbisDocRef();
+		result.setHasOrder(false);
+		String[] values = ibisDocRef.value();
+		String methodString = null;
+		if (values.length == 1) {
+			methodString = ibisDocRef.value()[0];
+		} else if (values.length == 2) {
+			methodString = ibisDocRef.value()[1];
+			try {
+				result.setOrder(Integer.parseInt(ibisDocRef.value()[0]));
+				result.setHasOrder(true);
+			} catch (Throwable t) {
+				log.warn(String.format("Could not parse order in @IbisDocRef annotation: "
+						+ Integer.parseInt(ibisDocRef.value()[0])));
+			}
+		}
+		else {
+			log.warn(String.format(String.format("Invalid @IbisDocRef annotation on method: %s.%s",
+					originalMethod.getDeclaringClass().getName(), originalMethod.getName())));
```

Comment:
Waarom is hij invalid? Laat dat weten in de foutmelding! Zoiets als: 'too many parameters in declaration'

Reply:
Gedaan.